diff --git a/config.def.h b/config.def.h
index ef6a251..77f8a20 100644
--- a/config.def.h
+++ b/config.def.h
@@ -47,7 +47,7 @@ static const Layout layouts[] = {
 };
 
 /* key definitions */
-#define MODKEY Mod1Mask
+#define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
 	{ MODKEY,                       KEY,      view,           {.ui = 1 << TAG} }, \
 	{ MODKEY|ControlMask,           KEY,      toggleview,     {.ui = 1 << TAG} }, \
@@ -60,7 +60,7 @@ static const Layout layouts[] = {
 /* commands */
 static char dmenumon[2] = "0"; /* component of dmenucmd, manipulated in spawn() */
 static const char *dmenucmd[] = { "dmenu_run", "-m", dmenumon, "-fn", dmenufont, "-nb", normbgcolor, "-nf", normfgcolor, "-sb", selbordercolor, "-sf", selfgcolor, NULL };
-static const char *termcmd[]  = { "st", NULL };
+static const char *termcmd[]  = { "dwm-terminal.sh", NULL };
 
 /*
  * Xresources preferences to load at startup
@@ -83,8 +83,36 @@ ResourcePref resources[] = {
 
 #include "patches/shiftview.c"
 
+/* reference: http://economical-with-words.blogspot.com/2013/02/taking-screenshots-in-dwm.html
+ */
+static const char *scrotthewholeshebangcmd[]  = { "scrot.sh", NULL };
+static const char *scrotregioncmd[]           = { "scrot.sh", "-s", NULL };
+static const char *scrotregionfreezecmd[]     = { "scrot.sh", "-s", "-f", NULL };
+static const char *scrotfocusedcmd[]          = { "scrot.sh", "--focused", NULL };
+static const char *scrotxclipcmd[]            = { "scrot-xclip.sh", "-s", NULL };
+
+#include <X11/XF86keysym.h>
+
+/* reference: https://gist.github.com/neuro-sys/2570361
+ * reference: https://tools.suckless.org/slock/patches/mediakeys/
+ * reference: https://hmagellan.com/blog/2020/aug/13-volume-dwm.html
+ */
+static const char *volupcmd[] = { "amixer", "-D", "default", "sset", "Master", "Playback", "5%+", NULL };
+static const char *voldowncmd[] = { "amixer", "-D", "default", "sset", "Master", "Playback", "5%-", NULL };
+static const char *voltogglecmd[] = { "amixer", "-D", "default", "sset", "Master", "Playback", "toggle", NULL };
+
 static Key keys[] = {
 	/* modifier                     key        function        argument */
+	{ 0,                            XK_Print,  spawn,          {.v = scrotthewholeshebangcmd } },
+	/* { ShiftMask,                    XK_Print,  spawn,          SHCMD("scrot.sh -s -f 2>&1 | /command/tai64n | tee -a /tmp/scrot.log") }, */
+	{ ShiftMask,                    XK_Print,  spawn,          {.v = scrotregioncmd } },
+	{ MODKEY|ShiftMask,             XK_Print,  spawn,          {.v = scrotregionfreezecmd } },
+	{ MODKEY,                       XK_Print,  spawn,          {.v = scrotfocusedcmd } },
+	{ ControlMask|ShiftMask,        XK_Print,  spawn,          {.v = scrotxclipcmd } },
+        { 0,                            XF86XK_AudioRaiseVolume,  spawn, {.v = volupcmd } },
+        { 0,                            XF86XK_AudioLowerVolume,  spawn, {.v = voldowncmd } },
+        { 0,                            XF86XK_AudioMute,         spawn, {.v = voltogglecmd } },
+
 	{ MODKEY,                       XK_p,      spawn,          {.v = dmenucmd } },
 	{ MODKEY|ShiftMask,             XK_Return, spawn,          {.v = termcmd } },
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
@@ -120,6 +148,8 @@ static Key keys[] = {
 	TAGKEYS(                        XK_7,                      6)
 	TAGKEYS(                        XK_8,                      7)
 	TAGKEYS(                        XK_9,                      8)
+	{ MODKEY,                       XK_Prior, shiftview,       {.i = -1 } },
+	{ MODKEY,                       XK_Next, shiftview,        {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_q,      quit,           {0} },
 };
 
diff --git a/config.def.h b/config.def.h
index 2bd75a6..89c4bbf 100644
--- a/config.def.h
+++ b/config.def.h
@@ -2,7 +2,7 @@
 
 /* appearance */
 static unsigned int borderpx  = 1;        /* border pixel of windows */
-static unsigned int snap      = 32;       /* snap pixel */
+static unsigned int snap      = 10;       /* snap pixel */
 static int showbar            = 1;        /* 0 means no bar */
 static int topbar             = 1;        /* 0 means bottom bar */
 static const char *fonts[]          = { "monospace:size=10" };
@@ -120,8 +120,8 @@ static Key keys[] = {
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
 	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
 	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
-	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.05} },
-	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.05} },
+	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.01} },
+	{ MODKEY,                       XK_l,      setmfact,       {.f = +0.01} },
 	{ MODKEY,                       XK_Return, zoom,           {0} },
 	{ MODKEY,                       XK_Tab,    view,           {0} },
 	{ MODKEY|ShiftMask,             XK_c,      killclient,     {0} },
@@ -133,8 +133,6 @@ static Key keys[] = {
 	{ MODKEY,                       XK_space,  setlayout,      {0} },
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
 	{ MODKEY|ShiftMask,             XK_f,      togglefullscr,  {0} },
-	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
-	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
 	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
 	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
diff --git a/config.def.h b/config.def.h
index 89c4bbf..359f64b 100644
--- a/config.def.h
+++ b/config.def.h
@@ -29,7 +29,7 @@ static const Rule rules[] = {
 	 */
 	/* class      instance    title       tags mask     isfloating   monitor */
 	{ "Gimp",     NULL,       NULL,       0,            1,           -1 },
-	{ "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 },
+	/* { "Firefox",  NULL,       NULL,       1 << 8,       0,           -1 }, */
 };
 
 /* layout(s) */
diff --git a/config.def.h b/config.def.h
index 359f64b..dc7f97f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -46,6 +46,9 @@ static const Layout layouts[] = {
 	{ ">M>",      centeredfloatingmaster },
 };
 
+/* https://dwm.suckless.org/customisation/windows_key/
+ */
+
 /* key definitions */
 #define MODKEY Mod4Mask
 #define TAGKEYS(KEY,TAG) \
diff --git a/config.def.h b/config.def.h
index 5d4dcae..86f8e50 100644
--- a/config.def.h
+++ b/config.def.h
@@ -104,6 +104,16 @@ static const char *volupcmd[] = { "amixer", "-D", "default", "sset", "Master", "
 static const char *voldowncmd[] = { "amixer", "-D", "default", "sset", "Master", "Playback", "5%-", NULL };
 static const char *voltogglecmd[] = { "amixer", "-D", "default", "sset", "Master", "Playback", "toggle", NULL };
 
+void
+focusswap(const Arg *arg)
+{
+	static int direction = 1;
+	Arg a = {0};
+	a.i = direction;
+	focusstack(&a);
+	direction *= -1;
+}
+
 static Key keys[] = {
 	/* modifier                     key        function        argument */
 	{ 0,                            XK_Print,  spawn,          {.v = scrotthewholeshebangcmd } },
@@ -121,6 +131,7 @@ static Key keys[] = {
 	{ MODKEY,                       XK_b,      togglebar,      {0} },
 	{ MODKEY,                       XK_j,      focusstack,     {.i = +1 } },
 	{ MODKEY,                       XK_k,      focusstack,     {.i = -1 } },
+	{ Mod1Mask,                     XK_Tab,    focusswap,      {0} },
 	{ MODKEY,                       XK_i,      incnmaster,     {.i = +1 } },
 	{ MODKEY,                       XK_d,      incnmaster,     {.i = -1 } },
 	{ MODKEY,                       XK_h,      setmfact,       {.f = -0.01} },
